File: springboot-jpa-example/src/main/java/org.bitbucket.poad1010.example.springboot.jpa/resource/MessageResource.java
package org.bitbucket.poad1010.example.springboot.jpa.resource;

import org.bitbucket.poad1010.example.springboot.jpa.request.CreateRequest;
import org.bitbucket.poad1010.example.springboot.jpa.request.UpdateRequest;
import org.bitbucket.poad1010.example.springboot.jpa.response.MessageResponse;

import javax.validation.constraints.NotNull;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

/**
 * Created by ken-yo on 2016/08/06.
 */
@Produces(MediaType.APPLICATION_JSON)
@Path("/message")
public interface MessageResource {
    @GET
    List<MessageResponse> list();

    @GET
    @Path("{id}")
    MessageResponse get(@NotNull @PathParam("id") String id);

    @Consumes(MediaType.APPLICATION_JSON)
    @POST
    MessageResponse create(CreateRequest request);

    @Consumes(MediaType.APPLICATION_JSON)
    @PUT
    MessageResponse update(UpdateRequest request);

    @Path("{id}")
    @DELETE
    void delete(@NotNull @PathParam("id") String id);
}

File: scala-example/spring-boot-example/src/main/scala/com/github/poad/scala/springboot/example/resource/MessageResource.scala
package com.github.poad.scala.springboot.example.resource

import java.util.Optional

import com.github.poad.scala.springboot.example.content.Message
import com.github.poad.scala.springboot.example.entity.MessageEntity
import javax.ws.rs.core.MediaType
import javax.ws.rs._
import org.springframework.stereotype.Component

@Component
@Produces(Array(MediaType.APPLICATION_JSON))
@Path("/message")
trait MessageResource {
  @GET
  def all(): java.util.List[Message]

  @GET
  @Path("{id}")
  def byId(@PathParam("id")  id: String): Message

  @Consumes(Array(MediaType.APPLICATION_JSON))
  @POST
  def create(message: Message): String

  @Consumes(Array(MediaType.APPLICATION_JSON))
  @PUT
  @Path("{id}")
  def update(@PathParam("id")  id: String, message: Message): Message

  @DELETE
  @Path("{id}")
  def deleteById(@PathParam("id")  id: String): Message

}
